src/modules/credential/userpass/path_users.rs

```diff
@@ -0,0 +1,251 @@
+use std::{collections::HashMap, sync::Arc, time::Duration};
+
+use serde::{Deserialize, Serialize};
+
+use super::{UserPassBackend, UserPassBackendInner};
+use crate::{
+    errors::RvError,
+    logical::{Backend, Field, FieldType, Operation, Path, PathOperation, Request, Response},
+    new_fields, new_fields_internal, new_path, new_path_internal,
+    storage::StorageEntry,
+    utils::{deserialize_duration, serialize_duration},
+};
+
+//const DEFAULT_MAX_TTL: Duration = Duration::from_secs(365*24*60*60 as u64);
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct UserEntry {
+    pub password_hash: String,
+    pub policies: Vec<String>,
+    #[serde(serialize_with = "serialize_duration", deserialize_with = "deserialize_duration")]
+    pub ttl: Duration,
+    #[serde(serialize_with = "serialize_duration", deserialize_with = "deserialize_duration")]
+    pub max_ttl: Duration,
+}
+
+impl Default for UserEntry {
+    fn default() -> Self {
+        Self {
+            password_hash: String::new(),
+            policies: Vec::new(),
+            ttl: Duration::from_secs(0),
+            max_ttl: Duration::from_secs(0),
+        }
+    }
+}
+
+impl UserPassBackend {
+    pub fn users_path(&self) -> Path {
+        let userpass_backend_ref1 = Arc::clone(&self.inner);
+        let userpass_backend_ref2 = Arc::clone(&self.inner);
+        let userpass_backend_ref3 = Arc::clone(&self.inner);
+
+        let path = new_path!({
+            pattern: r"users/(?P<username>\w[\w-]+\w)",
+            fields: {
+                "username": {
+                    field_type: FieldType::Str,
+                    required: true,
+                    description: "Username of the user."
+                },
+                "password": {
+                    field_type: FieldType::Str,
+                    required: false,
+                    description: "Password for this user."
+                },
+                "policies": {
+                    field_type: FieldType::Str,
+                    required: false,
+                    description: "Policies for this user."
+                },
+                "ttl": {
+                    field_type: FieldType::Int,
+                    default: 0,
+                    description: "TTL for this user."
+                },
+                "max_ttl": {
+                    field_type: FieldType::Int,
+                    default: 0,
+                    description: "TTL for this user."
+                }
+            },
+            operations: [
+                {op: Operation::Read, handler: userpass_backend_ref1.read_user},
```

Comment:
Why this needs to be cloned into 3 copies?

Reply:
> Why this needs to be cloned into 3 copies?

That's because it's limited by Rust's variable ownership mechanism...